bin/nbb

```diff
@@ -344,6 +339,24 @@ find_centos_7_ami_ids() {
   done
 }
 
+shutdown_instances() {
+  aws ec2 describe-instances --filters "Name=instance-state-name,Values=running" | jq -c -r '.Reservations[] .Instances[] .InstanceId' > instance_list
+  echo "You are about to terminate the following instances:"
+  cat instance_list
+  read -r -p "Are you sure? [y/N] " response
+  if [[ $response =~ ^([yY][eE][sS]|[yY])$ ]]
+  then
+    IFS=$'\n'  # make newlines the separator
+    for i in `cat ./instance_list`; do
+      echo "Shutting down $i"
+      aws ec2 terminate-instances --instance-ids $i
+    done
+  else
+    echo "Nothing shutdown, exiting..."
+  fi
+  unset IFS
```

Comment:
Suggestion: if the user passes in an argument like `--cleanup` when calling this command/function/wrapper, *automatically remove the generated instances list file for them* right here. Lets users keep their filesystem tidy if they don't care about having that point in time snapshot of state I mentioned above, and allows other users to keep that data if they so desire; all that's required is for the user to tell us how they want it. :smiley: